### PR TITLE
Officially replace sn25 with gazetteer in search location service and API

### PR DESCRIPTION
**Pull request by gjn**
This is the correpsonding PR to https://github.com/geoadmin/service-sphinxsearch/pull/261. It

- Replaces `sn25` origin with `gazetteer` origin from the official documentation
- Replaces `sn25` with `gazetteer` in JS API
- Removes sn25 from search service. Using [`sn25` as origins parameter](http://mf-chsdi3.dev.bgdi.ch/gjn_sn25/rest/services/inspire/SearchServer?searchText=wasser&type=locations&origins=sn25) will result in a 400 Bad Request error.

[Testlink to documentation](http://mf-chsdi3.dev.bgdi.ch/gjn_sn25/services/sdiservices.html#search)

### CODE REVIEW

File: chsdi/views/search.py [hunk 1]
```diff
@@ -327,7 +327,6 @@ def _origins_to_ranks(self, origins):
             'gg25': 2,
             'district': 3,
             'kantone': 4,
-            'sn25': 5,
```
Comment (loicgasser) [inline chsdi/views/search.py]: It may be used, but ok if we remove the indices from the search server.

**Comment by loicgasser** [comment]
LGTM

**Comment by loicgasser** [comment]
Build error apparently related to zeitreihen update

**Comment by gjn** [comment]
@ltclm @daguer Regarding the zeitreihen data change: quite a big differense between

[now](http://api3.geo.admin.ch/rest/services/all/MapServer/ch.swisstopo.zeitreihen/releases?geometry=650000.0%2C170000.0&imageDisplay=500%2C600%2C96&geometryType=esriGeometryPoint&mapExtent=611399.9999999999%2C158650%2C690299.9999999999%2C198150;) and [before](http://mf-chsdi3.dev.bgdi.ch/rest/services/all/MapServer/ch.swisstopo.zeitreihen/releases?geometry=650000.0%2C170000.0&imageDisplay=500%2C600%2C96&geometryType=esriGeometryPoint&mapExtent=611399.9999999999%2C158650%2C690299.9999999999%2C198150;)

Is this ok?

**Comment by ltclm** [comment]
not sure if everything is ok.
[dev](http://mf-chsdi3.dev.bgdi.ch/rest/services/all/MapServer/ch.swisstopo.zeitreihen/releases?geometry=650000.0%2C170000.0&imageDisplay=500%2C600%2C96&geometryType=esriGeometryPoint&mapExtent=611399.9999999999%2C158650%2C690299.9999999999%2C198150;)
db request and response:
```sql
$ psql -c "SELECT produkt,kbnum,kbbez,release_year,bgdi_order, array_upper(release_year,1) FROM public.meta_15 WHERE ST_DWITHIN( the_geom, ST_GeomFromWKB('\x01010000000000000020d623410000000080c00441'::bytea, 21781), 0.0) ORDER BY  bgdi_order" -d zeitreihen_master
 produkt | kbnum |           kbbez           |                                      release_year                                      | bgdi_order | array_upper
---------+-------+---------------------------+----------------------------------------------------------------------------------------+------------+-------------
 lk25    | 1209  | Brienz                    | {2012,2006,1998,1993,1987,1981,1975,1966}                                              |          1 |           8
 lk25    | 1229  | Grindelwald               | {2012,2006,1998,1993,1987,1981,1979,1976,1973}                                         |          1 |           9
 lk50    | 254   | Interlaken                | {2012,2006,1998,1993,1987,1981,1980,1971,1962,1959,1956,1949,1941,1938}                |          2 |          14
 lk100   | 37    | Brünigpass                | {2006,1998,1993,1987,1981,1976,1970,1969,1957}                                         |          3 |           9
 ta50    | 396   | Grindelwald               | {1930,1927,1913,1899,1894,1890,1870}                                                   |          5 |           7
 ta50    | 392   | Brienz                    | {1930,1928,1920,1916,1912,1905,1896,1889,1870}                                         |          5 |           9
 tk100   | 13    | Interlaken, Sarnen, Stans | {1934,1928,1919,1912,1909,1907,1905,1901,1900,1895,1893,1889,1883,1875,1871,1869,1864} |          6 |          17
(7 rows)
```
[prod](http://api3.geo.admin.ch/rest/services/all/MapServer/ch.swisstopo.zeitreihen/releases?geometry=650000.0%2C170000.0&imageDisplay=500%2C600%2C96&geometryType=esriGeometryPoint&mapExtent=611399.9999999999%2C158650%2C690299.9999999999%2C198150;)
db request and response
```sql
~$ psql -c "SELECT produkt,kbnum,kbbez,release_year,bgdi_order, array_upper(release_year,1) FROM public.meta_15 WHERE ST_DWITHIN( the_geom, ST_GeomFromWKB('\x01010000000000000020d623410000000080c00441'::bytea, 21781), 0.0) ORDER BY  bgdi_order" -d zeitreihen_prod
 produkt | kbnum |           kbbez           |                                      release_year                                      | bgdi_order | array_upper
---------+-------+---------------------------+----------------------------------------------------------------------------------------+------------+-------------
 lk25    | 1229  | Grindelwald               | {2006,1998,1993,1987,1981,1979,1976,1973}                                              |          1 |           8
 lk25    | 1209  | Brienz                    | {2006,1998,1993,1987,1981,1975,1966}                                                   |          1 |           7
 lk50    | 254   | Interlaken                | {2006,1998,1993,1987,1981,1980,1971,1962,1959,1956,1949,1941,1938}                     |          2 |          13
 lk100   | 37    | Brünigpass                | {2006,1998,1993,1987,1981,1976,1970,1969,1957}                                         |          3 |           9
 ta50    | 392   | Brienz                    | {1930,1928,1920,1916,1912,1905,1896,1889,1870}                                         |          5 |           9
 ta50    | 396   | Grindelwald               | {1930,1927,1913,1899,1894,1890,1870}                                                   |          5 |           7
 tk100   | 13    | Interlaken, Sarnen, Stans | {1934,1928,1919,1912,1909,1907,1905,1901,1900,1895,1893,1889,1883,1875,1871,1869,1864} |          6 |          17
```
there are 3 new releases in this extent:
lk25->1209->2012
lk25->1229->2012
lk50->254->2012

but the order of the result has changed too (see results with the same bgdi_order).
I think change is responsible for the big difference in the older years.
I will add an order by kbnum to the view definition. this should stabilize the result order.

**Comment by ltclm** [comment]
probably this code has to be improved too:
https://github.com/geoadmin/mf-chsdi3/blob/d81e52fe864546e2e9f39b64e9d22087f0e6ef5c/chsdi/views/features.py#L560

Im not sure that merging the arrays is working like that, it depends on the order of the input features...

**Comment by davidoesch** [comment]
@gjn  we can close this?

**Pull request merged**